[plugin-web-app] Add step closes current window with alerts handling (#2620)
vividus-framework/vividus: docs/modules/plugins/pages/plugin-web-app.adoc

When I open URL `https://docs.vividus.dev` in new window
----

=== Close current window

> Closes current window and switches to the previous window.

Reviewer: ```suggestion
Closes the current window and switches to the previous window.
```

[source,gherkin]
----
When I close the current window
----

IMPORTANT: Handling {popup-alert}[alerts] displayed with '{onbeforeunload-event}' events is not implied by the WebDriver specification to {close-spec} window. For handling alerts use step based on JavaScript '<<_close_current_window_with_alert_handling>>'.

WARNING: This step can only be applied to a session with multiple windows open.

.Open URL in new tab, close it and switch to the previous page
[source,gherkin]
----
Given I am on a page with the URL 'https://example.com/'
When I open URL `https://example.com/contact-us` in new window
When I close the current window
----

=== Close current window with alert handling

> Trying to close current window with '{onbeforeunload-event}' events handling.

Reviewer: ```suggestion
Trying to close the current window with '{onbeforeunload-event}' events handling.
```

> * If an {popup-alert}[alert] window via '{onbeforeunload-event}' event opens, it must be checked and handled in the next steps

Reviewer: ```suggestion
* If an {popup-alert}[alert] window via '{onbeforeunload-event}' event is opened, it must be checked and handled in the subsequent steps.
```

> * If an {popup-alert}[alert] window does not open, closes current window and switches to the window from which redirected to current window

Reviewer: ```suggestion
* If an {popup-alert}[alert] window is not opened, the step closes the current window and switches to the previous window.
```

[source,gherkin]
----

> When I try to close the current window with alerts handling

Reviewer: ```suggestion
When I attempt to close current window with possibility to handle alert
```

----

IMPORTANT: This step can only be used if the current window was opened via the step ``When I open URL \`$pageUrl` in new window``.

> NOTE: If you confirm window close in {popup-alert}[alert], the window will be closed, and you will need to switch to current window using the next step: `When I switch to window with title that $stringComparisonRule `$windowName``.

Reviewer: ```suggestion
NOTE: If you confirm window close in {popup-alert}[alert], the window will be closed, and you will need to switch to current window using the following step: `When I switch to window with title that $stringComparisonRule `$windowName``.
```
